site/docs/types/simple_logical_types.md

```diff
@@ -1,24 +1,22 @@
-# Simple Types
+# Simple Logical Types
 
-Substrait tries to cover the most common types used in data manipulation. Simple types are those that don't support any form of configuration. For simplicity, any generic type that has only a small number of discrete implementations is declared directly, as opposed to via configuration.
+Simple types are those that don't support any form of configuration. For simplicity, any generic type that has only a small number of discrete implementations is declared directly, as opposed to via configuration.
```

Comment:
As I've thought more about this, I think we should just remove the distinction between simple and compound types. Technically simple types also have properties: they are all configurable with nullability and type variation. Additionally, the number of times people have said "why don't you have x" when they were looking at one table and not the other suggests that the page separation is also confusing. Can you just collapse to a single page? (please include a forward from the old page to the new consolidated page for SEO purposes).

Reply:
I'm fine with collapsing the pages and tables into one, but I kind of like the distinction between simple and compound personally. To me, `i8` is a concrete type, while `struct` is not; the concrete form would be `struct<>`, but `i8<>` makes no sense. It's just a way to talk about parameterizable and non-parameterizable types that doesn't take 7-8 syllables, though.

Reply:
I think this is the root of the disconnect. I wouldn't agree that I8 is a concrete type. I8?[0] or I8[0] are concrete types. **All types are all parameterized**. Some of those parameters are common and some are unique. I think the confusion elsewhere stems from the fact that we don't say "I8 is a parameterized type" when it is (nullability and physical variation).

Reply:
I would just say `i8` is syntactic sugar for `i8[0]`. When someone is talking about `i8` and they're not specifically talking about type classes (which is almost never in practice, and quite clear from context normally), I assume they mean `i8[0]`. I *think* the only disconnect after that is that I would never refer to variations using integers. I can only assume you're referring to the protobuf value of `type_variation_reference` (otherwise I have no idea where those numbers are coming from), but AFAICT the only context in which type parsing is done is in YAML extensions, where anchor/reference pairs don't exist yet. In extensions they have names, just like functions and user-defined type classes. Talking about any number except 0 would also be wildly ambiguous, since plans can use any set of anchor numbers they want.

Assuming we can agree that `i8[0]` is represented using type variation reference 0, and that there is no type variation anchor defined for variation 0 in the extension set because it's the default/base/0/whatever variation, expanding the name would syntactically make it either `[]` or ` `, so the latter is what I've been doing to cut back on noise. Hence, to me, `i8` can both be used to refer to the type class `i8`, and to the concrete type non-nullable `i8` with the default variation. A type with a non-default variation could for example be `string[arrow]`.

Reply:
I agree that we should codify 0 if not defined is "system preferred" or similar.